### PR TITLE
fs-1987-add forms-option

### PR DESCRIPTION
**Pull request by SentientHousePlant**
- Added forms option to get on /applications 
- Restructured folders to match assessment store and support future development.

### CODE REVIEW

**Review by gio-karunakaran — APPROVED** [review]
Amazing refactor (although could belong in separate PR?) and love the forms option, very nice 🙌🏾

File: db/exc/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .application import ApplicationError
```
Comment (gidsg) [inline db/exc/__init__.py:1]: Please don't use abbreviations. Please can we rename this folder `exceptions` if that's what its for,

File: db/migrations/versions/5bf853808dfb_.py [hunk 2]
```diff
@@ -1,58 +1,85 @@
...
-    sa.Column('reference', sa.String(), nullable=False),
-    sa.Column('project_name', sa.String(), nullable=True),
-    sa.Column('started_at', sa.DateTime(), server_default=sa.text('now()'), nullable=True),
-    sa.Column('status', status_enum, nullable=False),
-    sa.Column('date_submitted', sa.DateTime(), nullable=True),
-    sa.Column('last_edited', sa.DateTime(), nullable=True),
-    sa.PrimaryKeyConstraint('id', name=op.f('pk_applications')),
-    sa.UniqueConstraint('fund_id', 'round_id', 'key', name='_reference'),
-    sa.UniqueConstraint('reference', name=op.f('uq_applications_reference'))
+    status_enum = postgresql.ENUM(
```
[the remark this answers is not in the document]
Comment (SentientHousePlant) [reply db/migrations/versions/5bf853808dfb_.py:22]: Thanks for spotting this, unsure why this migration was changed like this.

[the remark this answers is not in the document]
Comment (gidsg) [reply db/migrations/versions/5bf853808dfb_.py:22]: Ah, makes sense, although not sure why they were being linted if not being modified. Can we revert them?

**Review by gidsg — APPROVED** [review]
LGTM 👍 (Was quite hard to find the relevant changes amidst all the refactoring but I think it looks OK)

**Pull request merged**